Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,27 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+# Truncates a text string to fit within a byte size constraint,
+# preserving character encoding validity. The returned string will
+# be no longer than the specified max_bytes, though the exact
+# shortfall may vary.
+def truncate_text_to_approximate_size(s, max_bytes, options = {})
+  front_weight = options[:front_weight] || 0.5
```

Comment:
I wanted to put all the valid `options` at the top so you could read them as part of the function signature without having to read through the whole function for them.